.golangci.yml

```
@@ -4,31 +4,52 @@ run:
 
 linters:
   enable:
+    - gci
     - gofmt
-    - staticcheck
     - gosimple
-    - unused
-    - tparallel
     - paralleltest
     - prealloc
-    - unparam  # This was added while we used go1.18 which means it's always been disabled. It may eventually be re-enabled and we'll need to fix what it finds
+    - staticcheck
+    - tparallel
+    - unparam 
+    - unused
   presets:
     - bugs
     - unused
   disable:
     # The below rules are disabled due to bug with go1.18 and generics: https://github.com/golangci/golangci-lint/issues/2859
     - stylecheck
     - musttag # Extremely slow, at least on CI machines
+
 linters-settings:
-  govet:
-    check-shadowing: true
-  unused:
-    generated-is-used: true
   exhaustive:
     default-signifies-exhaustive: true
+  gci:
+    custom-order: true
+    sections:
+      - standard
+      - dot
+      - alias
+      - default
+      - prefix(github.com/Azure/azure-service-operator/)
```

I don't mind pulling out dot aliases because they're weird/uncommon and also have a different impact on import usage than standard impacts.

I'm not sold on pulling out just plain aliases though. Plain aliases are much more common and don't really change the usage of the package, nor do they really feel related to one another, at least not in nearly the same way that the standard/default/prefix groups are.

What do you think about just:

```suggestion
    sections:
      - standard
      - dot
      - default
      - prefix(github.com/Azure/azure-service-operator/)
```

---

I could be persuaded. I tend to pull the aliased imports out into a separate group because it looks messy with them in the `default` list.

---

I prefer them inline. I took a quick look at some random other projects and determined:
... a lot of projects don't enforce any ordering, which isn't surprising I guess because the default Go tooling doesn't either.

the ones I found that seemed to follow a pattern follow the one I suggested (not pulling the `alias` out and allowing inline aliases):
- [msal1](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/main/apps/confidential/confidential_test.go), [msal2](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/main/apps/internal/base/internal/storage/items_test.go)
- [controller-runtime1](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/envtest/envtest_suite_test.go), [controller-runtime2](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/cache/cache.go)
- [client go](https://github.com/kubernetes/client-go/blob/master/openapi/groupversion_test.go)

I probably prefer that pattern because it's the same pattern Python had (which also allows inline aliases).